server/src/main/java/org/elasticsearch/action/update/UpdateRequest.java

```diff
@@ -838,7 +809,6 @@ public void writeTo(StreamOutput out) throws IOException {
             doc.id(id);
             doc.writeTo(out);
         }
-        out.writeOptionalStringArray(fields);
```

Comment:
You'll want to do something like:
```
if (false == out.getVersion().onOrAfter(Version.V_7_0_0_alpha_1) {
  out.writeOptionalStringArray(null);
}
```

That way 6.x can parse the request. We know there aren't any `fields` anyway because we couldn't have parsed them.